onnx/defs/generator/defs.cc

```diff
@@ -51,6 +98,60 @@ ONNX_OPERATOR_SET_SCHEMA(
             return;
           }
 
+          if (nullptr != value_int) {
+            // OpSchema::Verify check ensures that the attribute value has_i():
+            if (!value_int->has_i())
+              fail_shape_inference("Attribute 'value_int' expect an integer.")
+            updateOutputElemType(ctx, 0, TensorProto::INT64);
+            appendDim(getOutputShape(ctx, 0), 1);
```

Comment:
IMO the scalar attributes should produce a scalar tensor, which should have zero dimensions https://github.com/onnx/onnx/blob/master/docs/IR.md#static-tensor-shapes